### PR TITLE
Disconnect adopted dry refactor

### PR DESCRIPTION
**Pull request by gubatron**
Just a warm-up DRY refactor after reading 0000500588d596b3f15a712fe001e7b5ae9d5fff

(Want to learn more C with hands on work on a project that matters, love how you've organized things and how the build process is painless.)

Cheers from FrostWire. We're wondering if cjdns could be used at the network layer, so that instead of binding our bittorrent (libtorrent) sockets to regular TCP/IP endpoints, we could instead bind them to cjdns IPv6 sockets. Still reading the basics about your project.

### CODE REVIEW

**Comment by gubatron** [comment]
hmm, not sure why my previous (already merged into crashey commit) got in here. Sorry about that.

**Comment by lgierth** [comment]
> hmm, not sure why my previous (already merged into crashey commit) got in here. Sorry about that.

that's just because you opened the PR against master instead of crashey

**Comment by kyrias** [comment]
> We're wondering if cjdns could be used at the network layer, so that instead of binding our bittorrent (libtorrent) sockets to regular TCP/IP endpoints, we could instead bind them to cjdns IPv6 sockets.

This question doesn't really make any sense. "cjdns IPv6 sockets" are just regular IPv6 sockets. If you support IPv6, you support cjdns.

**Comment by gubatron** [comment]
Thanks @kyrias, you've answered my non-sensical question (plz take it easy on noobs)

Thanks @lgierth

**Comment by gubatron** [comment]
@kyrias if I wanted to perform a test, what tool in cjdns do I need to invoke so that my OS will show me a cjdns IP when I do this:
```
ifconfig | grep inet6
	inet6 ::1 prefixlen 128 
	inet6 fe80::1%lo0 prefixlen 64 scopeid 0x1 
	inet6 fe80::62f8:1dff:feab:6922%en0 prefixlen 64 scopeid 0x4 
	inet6 2602:30a:2c88:db10:62f8:1dff:feab:6922 prefixlen 64 autoconf 
	inet6 2602:30a:2c88:db10:fdb2:9012:87df:4f3a prefixlen 64 autoconf temporary 
	inet6 2602:30a:2c88:db10::49 prefixlen 64 dynamic 
	inet6 fe80::3c07:74ff:fe7a:53bf%awdl0 prefixlen 64 scopeid 0x8
    inet6 fc??:??????????????? #will my cjdns IP be here somehow?
```

**Comment by kyrias** [comment]
@gubatron Have cjdroute running. And the cjdns IP will be the one that starts with `fc`, yes.

**Comment by gubatron** [comment]
@kyrias thank you sir.

**Comment by kpcyrd** [comment]
@gubatron also, `sudo cat /etc/cjdroute.conf | cjdroute --cleanconf | jq -r .ipv6`

File: memory/Allocator.c [hunk 1]
```diff
@@ -319,41 +319,29 @@ static void connect(struct Allocator_pvt* parent,
     child->parent = parent;
 }
 
+static int findAllocator(struct Allocator_pvt* target, struct Allocator_List** cpp)
```
Comment (cjdelisle) [inline memory/Allocator.c]: misleading name, this function actually *disconnects* the allocator from the linked list.
Thanks for your effort and sorry it took so long to address it.

**Comment by cjdelisle** [comment]
Thanks for your contribution, merged to crashey and it will go into master in the next "release".

**Comment by gubatron** [comment]
thank you so much! @cjdelisle

**Pull request closed without merging**